libraries/AP_Mount/AP_Mount_Xacti.cpp

```diff
@@ -0,0 +1,537 @@
+            }
+            break;
+        }
+
+        case MAV_MOUNT_MODE_SYSID_TARGET:{
+            MountTarget angle_target_rad {};
+            if (get_angle_target_to_sysid(angle_target_rad)) {
+                send_target_angles(angle_target_rad.pitch, angle_target_rad.yaw, angle_target_rad.yaw_is_ef);
+            }
+            break;
+        }
+
+        default:
+            // we do not know this mode so raise internal error
+            INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+            break;
+    }
+}
+
+// return true if healthy
+bool AP_Mount_Xacti::healthy() const
+{
+    // unhealthy until gimbal has been found and replied with firmware version info
+    if (!_initialised) {
+        return false;
+    }
+
+    // unhealthy if attitude information NOT received recently
+    if (AP_HAL::millis() - _last_current_attitude_quat_ms > 1000) {
+        return false;
+    }
+
+    // if we get this far return healthy
+    return true;
+}
+
+// take a picture.  returns true on success
+bool AP_Mount_Xacti::take_picture()
+{
+    if (_detected_modules[_instance].ap_dronecan == nullptr) {
+        return false;
+    }
+
+    // set SingleShot parameter
+    return _detected_modules[_instance].ap_dronecan->set_parameter_on_node(_detected_modules[_instance].node_id, XACTI_PARAM_SINGLESHOT, 0, &param_int_cb);
+}
+
+// start or stop video recording.  returns true on success
+// set start_recording = true to start record, false to stop recording
+bool AP_Mount_Xacti::record_video(bool start_recording)
+{
+    if (_detected_modules[_instance].ap_dronecan == nullptr) {
+        return false;
+    }
+
+    // set Recording parameter
+    return _detected_modules[_instance].ap_dronecan->set_parameter_on_node(_detected_modules[_instance].node_id, XACTI_PARAM_RECORDING, start_recording ? 1 : 0, &param_int_cb);
+}
+
+// set focus specified as rate, percentage or auto
+// focus in = -1, focus hold = 0, focus out = 1
+bool AP_Mount_Xacti::set_focus(FocusType focus_type, float focus_value)
+{
+    if (_detected_modules[_instance].ap_dronecan == nullptr) {
+        return false;
+    }
+
+    // convert focus type and value to parameter value
+    uint8_t focus_param_value;
+    switch (focus_type) {
+    case FocusType::RATE:
+    case FocusType::PCT:
+        // focus rate and percentage control not supported so simply switch to manual focus
+        // FocusMode of 0:Manual Focus
+        focus_param_value = 0;
+        break;
+    case FocusType::AUTO:
+        // FocusMode of 1:Single AutoFocus, 2:Continuous AutoFocus
+        focus_param_value = 2;
+        break;
+    default:
+        // unsupported forucs mode
+        return false;
+    }
+
+    // set FocusMode parameter
+    return _detected_modules[_instance].ap_dronecan->set_parameter_on_node(_detected_modules[_instance].node_id, XACTI_PARAM_FOCUSMODE, focus_param_value, &param_int_cb);
+}
+/*
```

Comment:
better to have #if 0 perhaps?

Reply:
I've added an implementation for these two methods now so they are no longer commented out.